### DISCUSSION TITLE
Machine doesn't get deactivated with dead heartbeat

### DISCUSSION THREAD

**Discussion by FuRi0n** [keygen-sh/community#60] [Q&A] [answered]
I'm having some problems with the heartbeat. I have a policy with heartbeats enabled, if a machine is activated but it never sends his first heartbeat it never gets deactivated, taking a slot on the license. If the machine manages to send his first heartbeat everything works correctly. Is there a way to make these dead machines to deactivate automatically?

**Comment by ezekg** [comment, marked as answer]
Hey @FuRi0n, if you enable `requireHeartbeat` on the policy, the machine will automatically start its heartbeat after it is activated. So if it fails to send another heartbeat within the allotted window, it will be deactivated. If `requireHeartbeat` is not set, then the machine will never be considered dead, since it never started its heartbeat monitor.

This should resolve the issue you're seeing.

**Reply by FuRi0n** [reply]
Hi, doesn't look like that works, this is the policy I'm using:
![imagen](https://user-images.githubusercontent.com/7131643/217009246-042edbd4-8499-4b92-b065-17fa295c12d0.png)

And this is my machine configuration:
![imagen](https://user-images.githubusercontent.com/7131643/217009509-0d1b860c-a82c-4f09-a834-07939c530abc.png)

As you can see the policy is configured with 2 minutes heartbeat and 1 minute resurrection period. The machine has been dead for over 4 minutes and it's still there, it never gets deleted unless I sent the first heartbeat, then it gets deleted correctly after 3 minutes.

**Reply by ezekg** [reply]
Can you please email support so that I can get more information on this machine, such as ID and account ID? It may take a few minutes for a machine to automatically be pruned after it has died.

support@keygen.sh
